app/code/Ometria/Core/Service/Product/Inventory.php

```diff
@@ -122,6 +163,49 @@ private function getMSISalableQuantity(ProductInterface $product)
         return $qty;
     }
 
+    /**
+     * @param ProductInterface $product
+     * @return float|null
+     * @throws InputException
+     * @throws LocalizedException
+     * @throws SkuIsNotAssignedToStockException
+     */
+    private function getMSIGlobalSalableQuantity(ProductInterface $product)
+    {
+        /** @var GetAssignedStockIdsBySku $getSalableQtyData */
+        $getAssignedStockIdsBySku = ObjectManager::getInstance()->get(GetAssignedStockIdsBySku::class);
+
+        /** @var GetStockItemConfigurationInterface $getStockItemConfiguration */
+        $getStockItemConfiguration = ObjectManager::getInstance()->get(GetStockItemConfigurationInterface::class);
+
+        /** @var GetProductSalableQtyInterface $getProductSalableQtyInterface */
+        $getProductSalableQty = ObjectManager::getInstance()->get(GetProductSalableQtyInterface::class);
+
+        $qty = 0.;
+        $stockIds = $getAssignedStockIdsBySku->execute($product->getSku());
+
+        foreach ($stockIds as $stockId) {
+            $stockItemConfiguration = $getStockItemConfiguration->execute($product->getSku(), $stockId);
+            $isManageStock = $stockItemConfiguration->isManageStock();
+
+            if (!$isManageStock) {
+                return null;
```

Comment:
should we return null here if the return type should be a number ?

if the null value here serve some specific purpose, a comment will be great 🙂

Reply:
@mkuzdowicz The return of null here is to cover the scenario where the product has 'manage stock' disabled. This is different from having zero stock. The actual return type of the function is `?float` to allow for this. Note the extension functions don't use Return Type Declarations to allow backwards compatibility for merchants running php <7. I will add a comment here to make this more clear.